src/Transition/CreateNewIDFUsingRulesV23_2_0.f90

```diff
@@ -581,16 +580,18 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                 OutArgs(139)='' ! new speed 10 2023 rated field
                 OutArgs(140:149)=InArgs(116:125)
                 ! But then only modify CurArgs based on the number of fields
-                IF (CurArgs >= 29) CurArgs = CurArgs + 2  ! this will always trigger for speed 1
-                IF (CurArgs >= 39) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 49) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 59) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 69) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 79) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 89) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 99) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 109) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
-                IF (CurArgs >= 119) CurArgs = CurArgs + 2  ! only do this speed if we have that many inputs
+                NewArgs = 4
+                IF (CurArgs >= 25) NewArgs = NewArgs + 2  ! this will always trigger for speed 1
+                IF (CurArgs >= 35) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 45) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 55) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 65) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 75) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 85) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 95) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 105) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                IF (CurArgs >= 115) NewArgs = NewArgs + 2  ! only do this speed if we have that many inputs
+                CurArgs = CurArgs + NewArgs
```

Comment:
OK, I can see how updating CurArgs (the original total of Alphas and Numbers) while processing a complex object could confuse subsequent logic, especially for extensible objects. This looks like a good fix. I did not test but it looks like this will work.